### ISSUE TITLE
update and license

### ISSUE THREAD

**Issue by rgleason** [ptulp/LogbookKonni-1.2#26]
Here is the best locTion to discuss

**Comment by leamas** [comment]
@ptulp:

OK, here we go. Before we can even mention a license, we need copyright statements. If anyone still wonders why, let we know.

Assuming we can agree on the need, copyright statements are usually put in each source file at the very beginning. They might look something like this in this case:

```
/**
 * Copyright (c) 2000-2007 by Nicolas Devillard.
 * Copyright (c) 2009 by Tim Post <tinkertim@gmail.com>
 * 
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

```

The open issues are the names claiming copyright. Your (Peter's) name is obvious. As for Del and Konnibe one way is to just omit them. Theoretically, this carries risks if anyone of these complains, but I don't see this coming. 

Do you?

**Comment by rgleason** [comment]
readme.txt says
License
The plugin code is licensed under the terms of the GPL v2 or, at your will, later.

so I understand about the need for copyright now.

**Comment by leamas** [comment]
@ptulp @delatbabel: We need some input from you (and only you) here. Would it be OK if we added a copyright statement like [above](https://github.com/ptulp/LogbookKonni_pi/issues/26#issuecomment-877667661) bur with these lines in the beginning instead:

Copyright (c) 2018-2021 Peter Tulp
Copyright (c) 2013-2018 Del  LastName


@delababel: Sorry, I don't find your last name...

**Comment by delatbabel** [comment]
Hrm, I have actually done very little of the overall development of this package. So I don't really feel comfortable having my name in the copyright statement. The original developer was "Konni" who is no longer active in the OpenCPN community. If you can find his details (maybe somewhere else in the source files) then probably best add that. Otherwise simply omit my name.

**Comment by leamas** [comment]
Fair enough. We need  Konni's details  i. e.,  his name. Anyone?

**Comment by leamas** [comment]
Wait... we have https://www.copyright.gov/circs/circ32.pdf. Basically, this boils down to that we could write something like
```
Copyright (c) 2012 Konnibe
Copyright (c) 2013-2018 delatbabel
Copyright (c) 2018-2021 Peter Tulp
```

@delatbabel: Would this be acceptable for you?

**Comment by delatbabel** [comment]
Yes that would be fine.  Note that I'm not in the USA. Copyright laws probably work very differently here, USA government circulars are of limited value outside of the USA (or on the high seas).

**Comment by leamas** [comment]
Well, I'm happy as long as you are. After all, you are a US subject :) 

@ptulp:  Acceptable for you? Without feedback from you, we could proceed without mentioning you in the copyright. It will still be valid.

**Comment by leamas** [comment]
>  After all, you are a US subject :) 

Oops, I misread. That said, I'll assume that copyright laws are similar enough in this respect. In  for example Sweden we have exactly the same rules.

**Comment by leamas** [comment]
Just to make myself clear: My plan is to make  PRs against at both @delatbabel's and @ptulp's repo once we have settlled the details after getting some input from @ptulp.

**Comment by ptulp** [comment]
I would suggest the following:

Copyright (c) 2011-2013 Konnibe
Copyright (c) 2013-2015 Del Edson
Copyright (c) 2015-2021 Peter Tulp

Are you going to add the copyright note to every file?

**Comment by leamas** [comment]
@delatbabel: OK with you?  I'm fine with Peter's suggestion.

My plan is to add the notice to each file under src/  (obviously not the icons). It might be that someone feels he hasn't really touched  some  file. In that case the correct solution is to drop the copyright from the notice. But this is easier to do once we have the PR in place.

Thoughts?

**Comment by ptulp** [comment]
Sounds good to me.

**Comment by delatbabel** [comment]
that's fine by me

**Comment by leamas** [comment]
ok, the ball is on my side. I'm sailing, so it might or might not take some time. But, I'm on it.

**Comment by leamas** [comment]
Made a PR against @delatbabel [here](https://github.com/delatbabel/LogbookKonni-1.2/pull/40). This one needs to be merged before I make the similar thing for @ptulp  since I I cannot fork the same base project twice.

**Comment by ptulp** [comment]
I will update my repository from delatbabel repository, so you don't have to create a PR  for mine.

**Comment by leamas** [comment]
great, thanks!

**Comment by rgleason** [comment]
alec, del and ptulp thank you for setting this pi back on course!

**Comment by ptulp** [comment]
Rick,

What is the status on your branch. 
Are all the path problems solved??

**Comment by rgleason** [comment]
I had about 15 file conflicts that I had to resolve. to merge delatbabel.  Which I did. 
When I compare my repository  master branch, with leamas copyright branch it shows no changes 
When I try to build locally following Install.md and using   win_deps.bat  first, then compiling under windows, I don't get any tarbball. 

Now sure what to do yet.

**Comment by rgleason** [comment]
https://github.com/rgleason/LogbookKonni_pi/commits/master
All builds are failing.

I think I am going to have to revert back to  July 10
Merge pull request #33 from leamas/master
Leamas updates and improvements

Later.

**Comment by rgleason** [comment]
The copyright and license version is not the most current version as far as I can determine.
Therefore I think I am going to have to unwind my master branch.

**Comment by leamas** [comment]
Don't care about this for now. I can help to merge these changes later.

**Comment by rgleason** [comment]
Made PR of my "v1.4.16.0 + SD AL Improvements-Repairs"  to delatbabel which has some conflicts which I might be able to resolve if given those rights. under delatbabels  repository, under options or settings. Otherwise Alec will need to merge it later when he gets back..
